app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java

```diff
@@ -542,7 +537,7 @@ public void uploadOfflineImages(Context context, boolean cancel, JobParameters j
         if (!cancel) {
 //            Toast.makeText(context, "called function", Toast.LENGTH_SHORT).show();
             task.job = job;
-            task.service = new WeakReference<SavedProductUploadJob>(service);
+            task.service = new WeakReference<>(service);
```

Comment:
Was this necessary? doesn't this has any consequences?

Reply:
This was unnecessary, that's why I removed it. And it runs perfectly fine.